src/js/game/hud/parts/buildings_toolbar.js

```diff
@@ -143,8 +143,25 @@ export class HUDBuildingsToolbar extends BaseHUDPart {
             return;
         }
 
-        this.root.soundProxy.playUiClick();
-        this.sigBuildingSelected.dispatch(metaBuilding);
-        this.onSelectedPlacementBuildingChanged(metaBuilding);
+        let previouslySelected = false;
+        for (const buildingId in this.buildingHandles) {
+            const handle = this.buildingHandles[buildingId];
+            if (handle.metaBuilding === metaBuilding) {
+                if (handle.index === this.lastSelectedIndex) {
+                    previouslySelected = true;
+                }
+                break;
+            }
+        }
+
+        const buildingPlacer = this.root.hud.parts.buildingPlacer;
+
+        if (previouslySelected && buildingPlacer.currentMetaBuilding.get()) {
+            buildingPlacer.abortPlacement();
```

Comment:
Instead of using abortPlacement directly, simply use `this.sigBuildingSelected.dispatch(null)` - The idea is that the HUD parts only interact via events and do not refer to other parts explicitely (I know I did break that myself at some points)

Reply:
I would refactor this a bit and instead of having a `previouslySelected` variable, simply set `metaBuilding = null` in line 151 - Then you don't need the whole if construct